src/v/cluster/cloud_metadata/uploader.cc

```diff
@@ -138,18 +143,57 @@ ss::future<error_outcome> uploader::upload_next_metadata(
       },
     };
 
+    vlog(clusterlog.debug, "Proceeding to controller snapshot upload");
     auto upload_controller_errc = co_await maybe_upload_controller_snapshot(
       manifest, lazy_as, retry_node);
     if (upload_controller_errc != error_outcome::success) {
         co_return upload_controller_errc;
     }
+    if (co_await term_has_changed(synced_term)) {
+        co_return error_outcome::term_has_changed;
+    }
+
+    vlog(clusterlog.debug, "Proceeding to offsets upload");
+    auto offsets_nt_cfg = _topic_table.get_topic_cfg(
+      model::kafka_consumer_offsets_nt);
+    if (offsets_nt_cfg.has_value()) {
+        std::vector<std::vector<ss::sstring>> uploaded_offset_paths(
+          offsets_nt_cfg->partition_count);
+        for (int i = 0; i < offsets_nt_cfg->partition_count; i++) {
+            offsets_upload_request req;
+            const auto& nt = model::kafka_consumer_offsets_nt;
+            req.offsets_ntp = model::ntp{nt.ns, nt.tp, model::partition_id{i}};
+            req.cluster_uuid = _cluster_uuid;
+            req.meta_id = manifest.metadata_id;
+            vlog(
+              clusterlog.debug,
+              "Requesting offsets upload of {}",
+              req.offsets_ntp);
+            auto reply = co_await _offsets_uploader->request_upload(req, 30s);
+            if (reply.ec != cluster::errc::success) {
+                vlog(
+                  clusterlog.debug,
+                  "Error while requesting offsets upload of {}",
+                  req.offsets_ntp);
+                continue;
+            }
+            uploaded_offset_paths[i] = std::move(reply.uploaded_paths);
+        }
+        manifest.offsets_snapshots_by_partition = std::move(
+          uploaded_offset_paths);
+    }
```

Comment:
given the suspension points above, is it fine to not check for term_change before modifying the manifest?

Reply:
It's fine because the only update to the manifest in the cloud happens below in `upload_manifest`. That said, a sync point isn't a bad idea.